pkg/version/version_test.go

```diff
@@ -15,6 +17,76 @@ var versionFixtureData = []string{
 	"v0.1.3",
 }
 
+func TestLatestReleaseInDir(t *testing.T) {
+	latest, err := LatestReleaseInDir("testdata/latest_releases")
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	assert.Equal(t, "testdata/latest_releases/suite_11.5.12.yml", latest)
+}
+
+func TestLatestReleaseInDirBadDirectoryError(t *testing.T) {
+	_, err := LatestReleaseInDir("doesnotexist")
+	if !assert.Error(t, err) {
+		return
+	}
+
+	assert.EqualError(
+		t,
+		err,
+		"could not read releases directory doesnotexist: "+
+			"open doesnotexist: no such file or directory",
+	)
+}
+
+func TestLatestReleaseInDirEmptydirectoryError(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "version_test")
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer os.RemoveAll(tempDir)
+
+	_, err = LatestReleaseInDir(tempDir)
+	if !assert.Error(t, err) {
+		return
+	}
+
+	assert.EqualError(
+		t,
+		err,
+		"could not find any release files in '"+tempDir+"'",
+	)
+}
+
+func TestLatestReleaseInDirBadSemverReleaseFileError(t *testing.T) {
+	_, err := LatestReleaseInDir("testdata/latest_releases_bad_semver")
+	if !assert.Error(t, err) {
+		return
+	}
+
+	assert.EqualError(
+		t,
+		err,
+		"could not parse semver from 'suite_3.4.yml' in "+
```

Comment:
I wonder if it's a good idea to assert on error values. I think it makes the tests and implementation coupled so that the tests are brittle. IIRC @jonahx once brought this up too... Would like to know what y'all think?

Reply:
I always have leaned on some type of matching on the error message since if you just match on the `err != nil`, you may have a different error condition than the one you anticipated and the code would pass green.

Reply:
Doesn't that work better if you use error codes? Then you can look for matching error code, and even if the error message changes the code is the same

Not sure it's worth adding error codes here FWIW though

Reply:
I love the idea of error codes

Reply:
yeah error codes could be useful but I am leaning that they may be overkill (or at least partially orthogonal) to the initial release.

Reply:
I think we should make them standard. They solve my major concern re: brittle

Reply:
I would love to see them become standard. makes it a LOT easier to troubleshoot. I know we're the end-users of this particular project - but if you took a quick review, how many error codes would we need to add to make it part of this project? ie how big of an ask is it to start using them